### PR TITLE
[backport] Simplify batch normalization with cuDNN

### PR DESCRIPTION
**Pull request by takagi**
Backport of #5568

### CODE REVIEW

File: chainer/functions/normalization/batch_normalization.py [hunk 1]
```diff
@@ -697,21 +565,36 @@ def can_use_cudnn(self, xp):
...
-        return arr.reshape(numpy.prod(arr.shape[0:-1]), -1, 1, 1)
-
-
 def _x_hat(x, mean, inv_std):
     x_mu = x - mean
     x_mu *= inv_std
     return x_mu
 
 
+def _chainerx_compute_axis(x_ndim, gamma_ndim, axis):
```
Comment (toslunar) [inline chainer/functions/normalization/batch_normalization.py]: Could you remove codes for ChainerX?

Comment (takagi) [reply chainer/functions/normalization/batch_normalization.py]: Removed!

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.
